test/functional/lib/tests.cpp

```diff
@@ -6243,23 +6219,36 @@ GenericXskWait(
     // On another thread, briefly delay execution to give the main test
     // thread a chance to begin waiting. Then, produce RX and TX.
     //
-    DELAY_INDICATE_THREAD_CONTEXT Ctx;
+    struct DELAY_INDICATE_THREAD_CONTEXT {
+        UCHAR *Payload;
+        SIZE_T PayloadLength;
+        MY_SOCKET *Xsk;
+        NET_IFINDEX IfIndex;
+        UINT32 QueueId;
+        BOOLEAN Rx;
+        BOOLEAN Tx;
+    } Ctx;
     Ctx.Payload = Payload;
     Ctx.PayloadLength = sizeof(Payload);
     Ctx.Xsk = &Xsk;
     Ctx.IfIndex = If.GetIfIndex();
     Ctx.QueueId = FnMpIf.GetQueueId();
     Ctx.Rx = Rx;
     Ctx.Tx = Tx;
-    CXPLAT_THREAD_CONFIG ThreadConfig {
-        0, 0, NULL, DelayIndicateFn, &Ctx
-    };
-    CXPLAT_THREAD AsyncThread;
-    TEST_TRUE(CXPLAT_SUCCEEDED(CxPlatThreadCreate(&ThreadConfig, &AsyncThread)));
-    auto DeleteThread = wil::scope_exit([&]
-    {
-        CxPlatThreadDelete(&AsyncThread);
-    });
+
+    CxPlatAsyncT<DELAY_INDICATE_THREAD_CONTEXT> Async([](DELAY_INDICATE_THREAD_CONTEXT *Ctx) {
```

Comment:
can the parameter be a const *ctx?

Reply:
It's possible if we add another constructor to CxPlatAsync. It won't compile if I just add const here.

Reply:
I wonder if we should just use lambda capture instead, which I managed to implement on my local branch. But, I don't know how to implement templated arguments for the lambda so you would not be able to pass a context pointer to it (does it really matter when we have captured the local vars)?

Reply:
hmm, this works for me locally:

```
    CxPlatAsyncT<const DELAY_INDICATE_THREAD_CONTEXT> Async([](const DELAY_INDICATE_THREAD_CONTEXT *Ctx) {
```

Reply:
oh silly me